apps/src/lib/levelbuilder/code-docs-editor/ProgrammingEnvironmentEditor.jsx

```diff
@@ -183,19 +181,12 @@ export default function ProgrammingEnvironmentEditor({
           style={styles.textInput}
         />
       </label>
-      <label>
-        Image
-        <Button
-          onClick={() => setUploadImageDialogOpen(true)}
-          text="Choose Image"
-          color="gray"
-          icon="plus-circle"
-        />
-        {programmingEnvironment.imageUrl && (
-          <span>{programmingEnvironment.imageUrl}</span>
-        )}
-      </label>
-
+      <ImageInput
+        initialImageUrl={programmingEnvironment.imageUrl}
+        updateImageUrl={imgUrl =>
+          updateProgrammingEnvironment('imageUrl', imgUrl)
```

Comment:
this is way up the chain and probably out of scope for this PR, but i'm having trouble following what all of the valid keys are that live in updateProgrammingEnvironment / programmingEnvironment / remainingProgrammingEnvironment / initialProgrammingEnvironment . one idea would be to document it by having this proptype list out the valid keys using `PropTypes.shape` here: https://github.com/code-dot-org/code-dot-org/blob/2129a10bc9c85fb6be4b5ea6e76c11453109a926/apps/src/lib/levelbuilder/code-docs-editor/ProgrammingEnvironmentEditor.jsx#L234 feel free to just tell me / point me to what I'm missing though :-)

Reply:
That's great feedback! I've been inconsistent about creating shapes for these objects across the various components, in part because they change and then I forget to update them. Now that these objects are pretty stable, I should go back and do this...in a separate PR :)